ground/src/main/java/com/google/android/ground/system/GoogleApiManager.kt

```diff
@@ -37,29 +38,30 @@ constructor(
    * Displays a dialog to install Google Play Services, if missing. Throws an error if install not
    * possible or cancelled.
    */
-  suspend fun installGooglePlayServices() {
+  suspend fun installGooglePlayServices(): Boolean {
     val status = googleApiAvailability.isGooglePlayServicesAvailable(context)
-    if (status == ConnectionResult.SUCCESS) return
+    if (status == ConnectionResult.SUCCESS) return true
 
     val requestCode = INSTALL_API_REQUEST_CODE
-    startResolution(status, requestCode, GooglePlayServicesMissingException())
-    getNextResult(requestCode)
+    return startResolution(status, requestCode)
   }
 
-  private fun startResolution(status: Int, requestCode: Int, throwable: Throwable) {
-    if (!googleApiAvailability.isUserResolvableError(status)) throw throwable
+  private suspend fun startResolution(status: Int, requestCode: Int): Boolean {
+    if (!googleApiAvailability.isUserResolvableError(status)) return false
 
-    activityStreams.withActivity {
-      googleApiAvailability.showErrorDialogFragment(it, status, requestCode) { throw throwable }
+    return try {
+      activityStreams.withActivity { activity ->
+        googleApiAvailability.showErrorDialogFragment(activity, status, requestCode, null)
+      }
+      getNextResult(requestCode)
+    } catch (e: Exception) {
+      Timber.e("Activity result was not successful for requestCode: $requestCode")
```

Comment:
When does `getNextResult()` return false, as opposed to an exception being thrown?

Reply:
https://github.com/google/ground-android/issues/723

internally

```kt
getActivityResults(requestCode).first()
```
This could throw error, hence try catch